### PR TITLE
TM-44 future financial estimates apim

### PR DESCRIPTION
**Pull request by jawwadbaig**
TM-44 future financial estimates APIM - View transfer finance details across financial years

### CODE REVIEW

**Review by ConorTill — APPROVED** [review]

File: src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs [hunk 1]
```diff
@@ -23,25 +24,39 @@ public GetFinancialBreakdownHandler(IForecastingApiClient<ForecastingApiConfigur
...
-                AmountPledged = pledgesTask.Result.Pledges.Where(p => p.Status != PledgeStatus.Closed).Sum(x => x.Amount)
+                Commitments = breakdownTask.Breakdown.Sum(x => x.FundsOut.Commitments),
+                ApprovedPledgeApplications = breakdownTask.Breakdown.Sum(x => x.FundsOut.ApprovedPledgeApplications),
+                TransferConnections = breakdownTask.Breakdown.Sum(x => x.FundsOut.TransferConnections),
+                AcceptedPledgeApplications = breakdownTask.Breakdown.Sum(x => x.FundsOut.AcceptedPledgeApplications),
+                PledgeOriginatedCommitments = breakdownTask.Breakdown.Sum(x => x.FundsOut.PledgeOriginatedCommitments),                
+                FundsIn = breakdownTask.Breakdown.Sum(x => x.FundsIn),
+                ProjectionStartDate = breakdownTask.ProjectionStartDate,
+                NumberOfMonths = breakdownTask.NumberOfMonths,
+                CurrentYearEstimatedCommittedSpend = (breakdownTask.Breakdown.Sum(x => x.FundsOut.ApprovedPledgeApplications) +
```
Comment (chrisfoster76) [inline src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs:47]: As the formula for calculating these values is the same for each financial year, we have a lot of duplicated code here and I can't help but think we could de-dupe it and improve the readability.

File: src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs [hunk 2]
```diff
@@ -23,25 +24,39 @@ public GetFinancialBreakdownHandler(IForecastingApiClient<ForecastingApiConfigur
...
-                                                        (new GetPledgesRequest(request.AccountId));
+            var breakdownTask = await _forecastingApiClient.Get<GetTransferFinancialBreakdownResponse>
+                                                        (new GetTransferFinancialBreakdownRequest(request.AccountId, DateTime.UtcNow));
 
-            await Task.WhenAll(transferFinancialBreakdownTask, pledgesTask);
+            var nextYearBreakdowntask = await _forecastingApiClient.Get<GetTransferFinancialBreakdownResponse>
+                                                        (new GetTransferFinancialBreakdownRequest(request.AccountId, DateTime.UtcNow.AddYears(1)));
 
+            var yearAfterNextYearBreakdowntask = await _forecastingApiClient.Get<GetTransferFinancialBreakdownResponse>
+                                                        (new GetTransferFinancialBreakdownRequest(request.AccountId, DateTime.UtcNow.AddYears(2)));
```
Comment (chrisfoster76) [inline src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs:34]: The tasks are being awaited separately and could be awaited together to improve performance.

File: src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs [hunk 3]
```diff
@@ -23,25 +24,39 @@ public GetFinancialBreakdownHandler(IForecastingApiClient<ForecastingApiConfigur
...
-                FundsIn = transferFinancialBreakdownTask.Result.Breakdown.Sum(x => x.FundsIn),
-                ProjectionStartDate = transferFinancialBreakdownTask.Result.ProjectionStartDate,
-                NumberOfMonths = transferFinancialBreakdownTask.Result.NumberOfMonths,
-                AmountPledged = pledgesTask.Result.Pledges.Where(p => p.Status != PledgeStatus.Closed).Sum(x => x.Amount)
+                Commitments = breakdownTask.Breakdown.Sum(x => x.FundsOut.Commitments),
+                ApprovedPledgeApplications = breakdownTask.Breakdown.Sum(x => x.FundsOut.ApprovedPledgeApplications),
+                TransferConnections = breakdownTask.Breakdown.Sum(x => x.FundsOut.TransferConnections),
+                AcceptedPledgeApplications = breakdownTask.Breakdown.Sum(x => x.FundsOut.AcceptedPledgeApplications),
+                PledgeOriginatedCommitments = breakdownTask.Breakdown.Sum(x => x.FundsOut.PledgeOriginatedCommitments),                
+                FundsIn = breakdownTask.Breakdown.Sum(x => x.FundsIn),
```
Comment (chrisfoster76) [inline src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs:44]: Does FundsIn surface in the UI? If not, we shouldn't return it

File: src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs (hunk 2 above)
Comment (jawwadbaig) [reply src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs:34]: I have added await Task.WhenAll

File: src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs (hunk 1 above)
Comment (jawwadbaig) [reply src/SFA.DAS.ManageApprenticeships/Application/Queries/Transfers/GetFinancialBreakdown/GetFinancialBreakdownHandler.cs:47]: yes sure, duplication has been removed.

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**